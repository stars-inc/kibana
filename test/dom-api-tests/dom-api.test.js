/**
 * @jest-environment jsdom
 */


describe('fake dom api testing', () => {

  const { fetchingData } = require('../../samples/dom/app')

  // NOTE: with fake api jsdom
  const root = document.createElement('div')

  it('dom node is defined', () => {
    expect(root).toBeDefined()
  })

  // to load diff, change some keys in ./data/data.json
  it('match difference recieved data from data.json', () => {
    fetchingData('http://localhost:4444')
      .then(data => 
        Object.keys(data).forEach(item => expect(item).toMatchSnapshot())
      )
  })
})