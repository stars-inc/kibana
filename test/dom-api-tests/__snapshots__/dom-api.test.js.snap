// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fake dom api testing 1 1`] = `"agent"`;

exports[`fake dom api testing 1 2`] = `"bytes"`;

exports[`fake dom api testing 1 3`] = `"clientip"`;

exports[`fake dom api testing 1 4`] = `"extension"`;

exports[`fake dom api testing 1 5`] = `"geo"`;

exports[`fake dom api testing 1 6`] = `"host"`;

exports[`fake dom api testing 1 7`] = `"index"`;

exports[`fake dom api testing 1 8`] = `"ip"`;

exports[`fake dom api testing 1 9`] = `"machine"`;

exports[`fake dom api testing 1 10`] = `"memory"`;

exports[`fake dom api testing 1 11`] = `"message"`;

exports[`fake dom api testing 1 12`] = `"phpmemory"`;

exports[`fake dom api testing 1 13`] = `"refer"`;

exports[`fake dom api testing 1 14`] = `"request"`;

exports[`fake dom api testing 1 15`] = `"response"`;

exports[`fake dom api testing 1 16`] = `"tags"`;

exports[`fake dom api testing 1 17`] = `"timestamp"`;

exports[`fake dom api testing 1 18`] = `"url"`;

exports[`fake dom api testing 1 19`] = `"utc_time"`;

exports[`fake dom api testing 1 20`] = `"event"`;

exports[`fake dom api testing match difference recieved data from data.json 1`] = `"agent"`;

exports[`fake dom api testing match difference recieved data from data.json 2`] = `"bytes"`;

exports[`fake dom api testing match difference recieved data from data.json 3`] = `"clientip"`;

exports[`fake dom api testing match difference recieved data from data.json 4`] = `"extension"`;

exports[`fake dom api testing match difference recieved data from data.json 5`] = `"geo"`;

exports[`fake dom api testing match difference recieved data from data.json 6`] = `"host"`;

exports[`fake dom api testing match difference recieved data from data.json 7`] = `"index"`;

exports[`fake dom api testing match difference recieved data from data.json 8`] = `"ip"`;

exports[`fake dom api testing match difference recieved data from data.json 9`] = `"machine"`;

exports[`fake dom api testing match difference recieved data from data.json 10`] = `"memory"`;

exports[`fake dom api testing match difference recieved data from data.json 11`] = `"message"`;

exports[`fake dom api testing match difference recieved data from data.json 12`] = `"phpmemory"`;

exports[`fake dom api testing match difference recieved data from data.json 13`] = `"refer"`;

exports[`fake dom api testing match difference recieved data from data.json 14`] = `"request"`;

exports[`fake dom api testing match difference recieved data from data.json 15`] = `"response"`;

exports[`fake dom api testing match difference recieved data from data.json 16`] = `"tags"`;

exports[`fake dom api testing match difference recieved data from data.json 17`] = `"timestamp"`;

exports[`fake dom api testing match difference recieved data from data.json 18`] = `"url"`;

exports[`fake dom api testing match difference recieved data from data.json 19`] = `"utc_time"`;

exports[`fake dom api testing match difference recieved data from data.json 20`] = `"event"`;
